### PR TITLE
cherry pick pr #111557 to 1.23 :update smd to 4.2.3

### LINKED ISSUE #111819: k8s apiserver write big custom object timeout, apiserver high cpu
### What happened?

A timeout occurred when writing a large object

schema.yaml content size 180k
```
✦ ➜ kubectl apply -f schema.yaml
Error from server (Timeout): error when creating "schema.yaml": Timeout: request did not complete within requested timeout - context deadline exceeded
```

k8s apiserver log:
```
{"level":"warn","ts":"2022-08-12T12:54:22.608Z","logger":"etcd-client","caller":"v3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc0166e5dc0/#initially=[https://[::1]:2379]","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
E0812 12:54:22.608199       1 finisher.go:175] FinishRequest: post-timeout activity - time-elapsed: 52.630776167s, panicked: false, err: context deadline exceeded, panic-reason: <nil>
```

But it is normal when I test writing directly with etcdctl
```
cat schema.yaml | ./etcdctl --endpoints=https://[::1]:2379 --cacert=/etc/kubernetes/pki/etcd/ca.crt --cert=/etc/kubernetes/pki/apiserver-etcd-client.crt --key=/etc/kubernetes/pki/apiserver-etcd-client.key put /test
```

### What did you expect to happen?

normal

### How can we reproduce it (as minimally and precisely as possible)?

Write large objects normally

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
✦ ➜ kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.9", GitCommit:"c1de2d70269039fe55efb98e737d9a29f9155246", GitTreeState:"clean", BuildDate:"2022-07-13T14:26:51Z", GoVersion:"go1.17.11", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.9", GitCommit:"c1de2d70269039fe55efb98e737d9a29f9155246", GitTreeState:"clean", BuildDate:"2022-07-13T14:19:57Z", GoVersion:"go1.17.11", Compiler:"gc", Platform:"linux/amd64"}

</details>


### Cloud provider

<details>

</details>


### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
Linux xxxxx 5.4.119-1-tlinux4-0009.3 #1 SMP Thu Apr 28 09:35:59 CST 2022 x86_64 x86_64 x86_64 GNU/Linux

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by silenceper**
#### What type of PR is this?
cherry pick pr #111557 to k8s 1.23 :update smd to 4.2.3

#### Which issue(s) this PR fixes:

Fixes #111819

#### Special notes for your reviewer:
Run:
```
make generated_files 
hack/pin-dependency.sh sigs.k8s.io/structured-merge-diff/v4 v4.2.3
hack/update-internal-modules.sh
hack/update-vendor.sh
```

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -39,6 +39,7 @@ require (
 	github.com/emicklei/go-restful v2.9.5+incompatible
 	github.com/evanphx/json-patch v4.12.0+incompatible
 	github.com/fsnotify/fsnotify v1.4.9
+	github.com/go-openapi/spec v0.20.6
```
Comment (aojea) [inline go.mod:42]: is this dep ok @liggitt

**Comment by aojea** [comment]
/lgtm cancel

I didn't see the additional dep, it is better if Jordan review this

File: go.mod (hunk 1 above)
Comment (liggitt) [reply go.mod:42]: definitely not (and it's not actually required... the CI verification job is showing it as removed when vendor is regenerated)

make sure you have up-to-date generated files (`make generated`) to make sure you don't have old stale references to this dep in generated openapi go files, then run `hack/update-vendor.sh` again

Comment (silenceper) [reply go.mod]: thanks 
Run  this again:
```
make generated_files 
hack/pin-dependency.sh sigs.k8s.io/structured-merge-diff/v4 v4.2.3
hack/update-internal-modules.sh
hack/update-vendor.sh
```

File: go.mod [hunk 2]
```diff
@@ -255,6 +255,7 @@ replace (
 	github.com/go-logr/zapr => github.com/go-logr/zapr v1.2.0
 	github.com/go-openapi/jsonpointer => github.com/go-openapi/jsonpointer v0.19.5
 	github.com/go-openapi/jsonreference => github.com/go-openapi/jsonreference v0.19.5
+	github.com/go-openapi/spec => github.com/go-openapi/spec v0.20.6
```
Comment (liggitt) [inline go.mod:258]: this is unexpected... make sure you don't have stale generated openapi files (run `make clean generated_files`  before running `hack/update-vendor.sh`)

**Comment by liggitt** [comment]
/close
in favor of https://github.com/kubernetes/kubernetes/pull/111914

**Pull request closed without merging**